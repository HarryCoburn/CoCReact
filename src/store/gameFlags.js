const Flags = {
  MAX_TOLERANCE_LEVEL: 20,
  MAX_MORALSHIFTER_LEVEL: 10,
  MAX_DESIRES_LEVEL: 10,
  MAX_ENDURANCE_LEVEL: 10,
  MAX_MYSTICALITY_LEVEL: 10,
  MAX_WISDOM_LEVEL: 5,
  MAX_FORTUNE_LEVEL: -1, //no max level
  MAX_VIRILITY_LEVEL: 15,
  MAX_FERTILITY_LEVEL: 15,
  NEW_GAME_PLUS_RESET_CLIT_LENGTH_MAX: 1.5,
  NEW_GAME_PLUS_LEVEL: 0,
  NEW_GAME_PLUS_BONUS_STORED_XP: 0,
  NEW_GAME_PLUS_BONUS_STORED_ITEMS: null,
  INGNAM_WEAPONSMITH_TALKED: 0,
  INGNAM_ALCHEMIST_TALKED: 0,
  INGNAM_BLACKMARKET_TALKED: 0,
  INGNAM_RUMORS: 0
};

export default Flags;
