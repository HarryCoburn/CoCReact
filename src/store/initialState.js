import React from "react";
import { Menus } from "../scenes/menus.js";

// And we need an initial state
const initialState = {
  output: (
    <>
      <p>CoC Engine: Clean Version</p>
      <p>
        Original concept by Fenoxo and crew
        <br />
        Converted to JS/React by Matraia
      </p>
      <p>Version extremely early.</p>
      <p>Click on New Game to Start </p>
    </>
  ),
  day: "0", //to reduce
  hour: "12", //to reduce
  minute: "00", //to reduce
  UI: {
    showStats: false,
    showMenuBar: true
  },
  stats: {
    byID: {
      strength: {
        id: "strength",
        name: "Strength",
        value: 100,
        max: 100,
        min: 0
      },
      toughness: {
        id: "toughness",
        name: "Toughness",
        value: 100,
        max: 100,
        min: 0
      },
      speed: {
        id: "speed",
        name: "Speed",
        value: 100,
        max: 100,
        min: 0
      },
      intelligence: {
        id: "intelligence",
        name: "Intelligence",
        value: 100,
        max: 100,
        min: 0
      },
      libido: {
        id: "libido",
        name: "Libido",
        value: 100,
        max: 100,
        min: 0
      },
      sensitivity: {
        id: "sensitivity",
        name: "Sensitivity",
        value: 100,
        max: 100,
        min: 0
      },
      corruption: {
        id: "corruption",
        name: "Corruption",
        value: 100,
        max: 100,
        min: 0
      },
      hp: {
        id: "hp",
        name: "HP",
        value: 100,
        max: 100,
        min: 0
      },
      lust: {
        id: "lust",
        name: "Lust",
        value: 100,
        max: 100,
        min: 0
      },
      fatigue: {
        id: "fatigue",
        name: "Fatigue",
        value: 100,
        max: 100,
        min: 0
      },
      hunger: {
        id: "hunger",
        name: "Hunger",
        value: 100,
        max: 100,
        min: 0
      },
      level: {
        id: "level",
        name: "Level",
        value: 1,
        max: 99
      },
      xp: {
        id: "xp",
        name: "XP",
        value: 0,
        min: 0,
        max: 99
      },
      gems: {
        id: "gems",
        name: "Gems",
        value: 0,
        min: 0,
        max: 9999
      }
    },
    coreIDs: [
      "strength",
      "toughness",
      "speed",
      "intelligence",
      "libido",
      "sensitivity",
      "corruption"
    ],
    combatIDs: ["hp", "lust", "fatigue", "hunger"],
    advIDs: ["level", "xp", "gems"],
    allIDs: [
      "strength",
      "toughness",
      "speed",
      "intelligence",
      "libido",
      "sensitivity",
      "corruption",
      "hp",
      "lust",
      "fatigue",
      "hunger",
      "level",
      "xp",
      "gems"
    ]
  },
  buttons: {
    byID: {
      u1: Menus.main,
      u2: Menus.data
    },
    lowerIDs: [
      "b1",
      "b2",
      "b3",
      "b4",
      "b5",
      "b6",
      "b7",
      "b8",
      "b9",
      "b10",
      "b11",
      "b12",
      "b13",
      "b14",
      "b15"
    ],
    upperIDs: ["u1", "u2", "u3", "u4", "u5", "u6", "u7"]
  }
};

export default initialState;
