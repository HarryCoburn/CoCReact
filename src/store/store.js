import rootReducer from "../reducers/root";
import { createStore } from "redux";

// Now we make a store

const store = createStore(rootReducer);

export default store;
