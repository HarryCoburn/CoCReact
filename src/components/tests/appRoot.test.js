import React from "react";
import { Provider } from "react-redux";
import renderer from "react-test-renderer";
import AppRootComponent from "../appRoot";
import configureStore from "redux-mock-store";
import * as I from "../../store/initialState";

const mockStore = configureStore([]);

describe("App Root Component", () => {
  let store = mockStore({
    output: I.iOutput,
    UI: I.iUIState,
    stats: I.iStats,
    buttons: I.iButtons,
    appearance: I.iAppearance,
    time: I.iTime,
    engine: I.iEngineState
  });
  let component;

  xtest("App root rendered correctly", () => {
    component = renderer.create(
      <Provider store={store}>
        <AppRootComponent></AppRootComponent>
      </Provider>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
