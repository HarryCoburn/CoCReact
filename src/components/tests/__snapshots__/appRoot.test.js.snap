// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Root Component App root rendered correctly 1`] = `
<div>
  <div
    className="menuBar visible"
  >
    <button
      className="button tooltip"
      onClick={[Function]}
    >
      New Game
      <span
        className="tooltiptext bottom"
      >
        Start a new game.
      </span>
    </button>
    <button
      className="button tooltip"
      onClick={[Function]}
    >
      Data
      <span
        className="tooltiptext bottom"
      >
        Save or load your files.
      </span>
    </button>
    <button
      className="blankButton"
      id="u3"
    />
    <button
      className="blankButton"
      id="u4"
    />
    <button
      className="blankButton"
      id="u5"
    />
    <button
      className="blankButton"
      id="u6"
    />
    <button
      className="blankButton"
      id="u7"
    />
    <button
      className="blankButton"
      id="u8"
    />
  </div>
  <div
    className="statBar invisible"
  >
    <h2>
      Name:
    </h2>
    <h2>
      Core Stats
    </h2>
    <div
      className="statFrame"
    >
      <div
        className="statGradient"
        style={
          Object {
            "width": "0%",
          }
        }
      >
        <p
          className="statLabel"
        >
          Strength
          : 
          0
           
        </p>
      </div>
      <div
        className="statUpDown"
      />
    </div>
    <div
      className="statFrame"
    >
      <div
        className="statGradient"
        style={
          Object {
            "width": "0%",
          }
        }
      >
        <p
          className="statLabel"
        >
          Toughness
          : 
          0
           
        </p>
      </div>
      <div
        className="statUpDown"
      />
    </div>
    <div
      className="statFrame"
    >
      <div
        className="statGradient"
        style={
          Object {
            "width": "0%",
          }
        }
      >
        <p
          className="statLabel"
        >
          Speed
          : 
          0
           
        </p>
      </div>
      <div
        className="statUpDown"
      />
    </div>
    <div
      className="statFrame"
    >
      <div
        className="statGradient"
        style={
          Object {
            "width": "0%",
          }
        }
      >
        <p
          className="statLabel"
        >
          Intelligence
          : 
          0
           
        </p>
      </div>
      <div
        className="statUpDown"
      />
    </div>
    <div
      className="statFrame"
    >
      <div
        className="statGradient"
        style={
          Object {
            "width": "0%",
          }
        }
      >
        <p
          className="statLabel"
        >
          Libido
          : 
          0
           
        </p>
      </div>
      <div
        className="statUpDown"
      />
    </div>
    <div
      className="statFrame"
    >
      <div
        className="statGradient"
        style={
          Object {
            "width": "0%",
          }
        }
      >
        <p
          className="statLabel"
        >
          Sensitivity
          : 
          0
           
        </p>
      </div>
      <div
        className="statUpDown"
      />
    </div>
    <div
      className="statFrame"
    >
      <div
        className="statGradient"
        style={
          Object {
            "width": "0%",
          }
        }
      >
        <p
          className="statLabel"
        >
          Corruption
          : 
          0
           
        </p>
      </div>
      <div
        className="statUpDown"
      />
    </div>
    <h2>
      Combat Stats
    </h2>
    <div
      className="statFrame"
    >
      <div
        className="statGradient"
        style={
          Object {
            "width": "0%",
          }
        }
      >
        <p
          className="statLabel"
        >
          HP
          : 
          0
           
          / 100
        </p>
      </div>
      <div
        className="statUpDown"
      />
    </div>
    <div
      className="statFrame"
    >
      <div
        className="statGradient"
        style={
          Object {
            "width": "0%",
          }
        }
      >
        <p
          className="statLabel"
        >
          Lust
          : 
          0
           
          / 100
        </p>
      </div>
      <div
        className="statUpDown"
      />
    </div>
    <div
      className="statFrame"
    >
      <div
        className="statGradient"
        style={
          Object {
            "width": "0%",
          }
        }
      >
        <p
          className="statLabel"
        >
          Fatigue
          : 
          0
           
          / 100
        </p>
      </div>
      <div
        className="statUpDown"
      />
    </div>
    <div
      className="statFrame"
    >
      <div
        className="statGradient"
        style={
          Object {
            "width": "0%",
          }
        }
      >
        <p
          className="statLabel"
        >
          Hunger
          : 
          0
           
          / 100
        </p>
      </div>
      <div
        className="statUpDown"
      />
    </div>
    <h2>
      Advancement
    </h2>
    <div
      className="statFrame"
    >
      <div
        className="statGradient"
        style={
          Object {
            "width": "1%",
          }
        }
      >
        <p
          className="statLabel"
        >
          Level
          : 
          1
           
        </p>
      </div>
      <div
        className="statUpDown"
      />
    </div>
    <div
      className="statFrame"
    >
      <div
        className="statGradient"
        style={
          Object {
            "width": "0%",
          }
        }
      >
        <p
          className="statLabel"
        >
          XP
          : 
          0
           
          / 99
        </p>
      </div>
      <div
        className="statUpDown"
      />
    </div>
    <div
      className="statFrame"
    >
      <div
        className="statGradient"
        style={
          Object {
            "width": "0%",
          }
        }
      >
        <p
          className="statLabel"
        >
          Gems
          : 
          0
           
        </p>
      </div>
      <div
        className="statUpDown"
      />
    </div>
    <div
      className="timeDisplay"
    >
      <p>
        Day#: 
        0
        <br />
        Time: 
        12
        :
        00
      </p>
    </div>
    <div
      className="spriteDisplay"
    />
  </div>
  <div
    id="maintext"
  >
    <p>
      CoC Engine: Clean Version
    </p>
    <p>
      Original concept by Fenoxo and crew
      <br />
      Converted to JS/React by Matraia
    </p>
    <p>
      Version extremely early.
    </p>
    <p>
      Click on New Game to Start 
    </p>
  </div>
  <div
    className="gridContainer"
  >
    <div
      className="lowerGrid"
    >
      <button
        className="blankButton"
        id="b1"
      />
      <button
        className="blankButton"
        id="b2"
      />
      <button
        className="blankButton"
        id="b3"
      />
      <button
        className="blankButton"
        id="b4"
      />
      <button
        className="blankButton"
        id="b5"
      />
      <button
        className="blankButton"
        id="b6"
      />
      <button
        className="blankButton"
        id="b7"
      />
      <button
        className="blankButton"
        id="b8"
      />
      <button
        className="blankButton"
        id="b9"
      />
      <button
        className="blankButton"
        id="b10"
      />
      <button
        className="blankButton"
        id="b11"
      />
      <button
        className="blankButton"
        id="b12"
      />
      <button
        className="blankButton"
        id="b13"
      />
      <button
        className="blankButton"
        id="b14"
      />
      <button
        className="blankButton"
        id="b15"
      />
    </div>
  </div>
</div>
`;
